plugin-server/src/cdp/hog-executor.ts

```diff
@@ -145,25 +141,40 @@ export class HogExecutor {
             invocation.hogFunctionId
         ]
 
+        const baseInvocation: HogFunctionInvocation = {
+            id: invocation.id,
+            globals: invocation.globals,
+        }
+
+        const errorRes: HogFunctionInvocationResult = {
+            ...baseInvocation,
+            hogFunctionId: invocation.hogFunctionId,
+            teamId: invocation.teamId,
+            success: false,
+            error: 'Something went wrong',
+            logs: [],
+        }
+
+        if (!hogFunction) {
+            errorRes.error = `Hog Function with ID ${invocation.hogFunctionId} not found`
+            return errorRes
+        }
+
         if (!invocation.vmState || invocation.error) {
             // TODO: Maybe add a log as well?
-            return {
-                ...invocation,
-                success: false,
-                error: invocation.error ?? new Error('No VM state provided for async response'),
-                logs: [],
-            }
+            errorRes.error = invocation.error ?? new Error('No VM state provided for async response')
+            return errorRes
         }
         invocation.vmState.stack.push(convertJSToHog(invocation.vmResponse ?? null))
```

Comment:
Nit: I'd put the `const errorRes` inside the error conditions themselves. Most of the time this object is created just to be directly discarded.

Reply:
Fair